### ISSUE TITLE
Create a view of a PEP after filtering columns.

### ISSUE THREAD

**Issue by donaldcampbelljr** [pepkit/pephub#322]
If I have a large PEP and I'd like to filter them based on columns, how do I save that such that I can point to the view path and run my analysis on that specific selection? Right now, I believe I'd have to copy the filtered samples and then paste them into a new PEP. 

Could we add the ability to save the filtered samples as a new view such that they can be referenced directly via the API?

Is view actually the correct approach for this?

**Comment by khoroshevskyi** [comment]
![image](https://github.com/pepkit/pephub/assets/41573628/7aaa37d2-fbdf-4401-987c-ef27983e7c36)

e.g. I want to create View using this samples. So I guess we have add button: `Save view`. We already have all APIs for that

**Comment by nleroy917** [comment]
This is a great idea.

The difficulty is extracting out the rows since the filtered data is kind of hidden behind the data table abstraction. It is possible, however! After some digging, you can do something like this:
```tsx
<HotTable
  afterFilter={(k) => {
      console.log(hotRef.current?.hotInstance?.getData());
      // set some state to store the current version of the filtered data
  }}
/>
```
This would let us capture the current state of the table **after** it's been filtered. Then, we can render a button that lets people hit an endpoint with that version of the table

**Comment by nleroy917** [comment]
@donaldcampbelljr @khoroshevskyi do you guys have any ideas for what the UI/layout would look like?

**Comment by donaldcampbelljr** [comment]
Hmm, maybe place a "Save View" button next to the views?
![image](https://github.com/pepkit/pephub/assets/125581724/5c138856-e5dc-4406-b63e-3d221de39db5)

And that pops up a small window to name and save the view?

**Comment by sanghoonio** [comment]
I think a 'save as' button might make most sense

![image](https://github.com/pepkit/pephub/assets/7041154/631776c7-4be2-4b99-995e-a6e8b1e437ff)

**Comment by nleroy917** [comment]
Great idea. Should it be "Save as view?" Is there anything else we can save it as? The `Save as` terminology seems to indicate there are going to be multiple options...

Regardless I really like that, and I would expect that to open a modal that lets you type in the name of the view... then hit confirm

**Comment by sanghoonio** [comment]
Just double checked the earlier discussion and I didn't realize 'views' referred to an actual feature within a PEP, not saving the current changes as a new copy of the PEP. In that case I agree about the 'save as' verbiage

**Comment by donaldcampbelljr** [comment]
Hmmm, does it make sense to put the additional save button on the right and not near the views box? I guess we are putting all the buttons on the right hand side....

Will you edit/change the views in the view box? That's why I thought the button might be better nearer the views box.

**Comment by sanghoonio** [comment]
Yeah, I'll put it back left since we are not saving as a separate PEP as I first thought

**Comment by nleroy917** [comment]
I initially had it by the views box, since its a "Save as view" feature. But, I kinda like the button over there on the right since it groups all the same features together... Save, Save as... discard.. etc

**Comment by sanghoonio** [comment]
Another option

<img width="1512" alt="image" src="https://github.com/pepkit/pephub/assets/7041154/721df195-2bf6-4759-a235-0396089d0d89">

**Comment by sanghoonio** [comment]
Thoughts on this?

<img width="1512" alt="image" src="https://github.com/pepkit/pephub/assets/7041154/eb14f41a-c21f-42cb-ad98-1fd653ec3cc7">

**Comment by nleroy917** [comment]
yeah I think thats nice... What do we think? That plus button is only enabled when the table has been filtered? Or its always enabled and it lets you select rows in the table to create a view? We will also need a way to delete views...

So, maybe it should be a gear and it opens a modal where you can delete views, add new ones, etc

**Comment by nleroy917** [comment]
could it be made to be smaller? It kind of sticks out right now

**Comment by donaldcampbelljr** [comment]
I like it!

> So, maybe it should be a gear and it opens a modal where you can delete views, add new ones, etc

Yeah, this would be great.
I think a gear or the plus sign would work fine either way. The gear is consistent with the other buttons. The plus sign may look a  little 'cleaner' ?

**Comment by nleroy917** [comment]
I changed the endpoints over in `dev_sam` so we need to make sure that this did not break:

- [ ] `pephubclient`
- [ ] `bedbase`

**Comment by nleroy917** [comment]
Likely solved with release `v0.13.0`
